### ISSUE TITLE
Android and iOS: Works in emulator but not on physical device

### ISSUE THREAD

**Issue by foggy1** [TheWidlarzGroup/react-native-video#560]
Is there any reason this library would load fine in an emulator, even in release build, but would not render anything on a physical phone?  We are using URI's to load the images.

**Comment by jonoirwinrsa** [comment]
@foggy1 I had this issue too and I know a fix for android and I can only guess that the same applies for ios.. When in prod if using require(...) it can't locate the video file correctly.

Putting the (mp4) videos in `android/app/src/main/res/raw` and referencing using `source={{uri: '<filename_no_extension>'}}` worked in prod and debug.

I guess there is a folder in ios/ where you can do the same thing and it will make it work

Good luck & please let me know if you get it to work :)

**Comment by Amurmurmur** [comment]
@jonoirwinrsa
Doesnt work for me :/ Put a video inside res/raw/background.mp4
Have the following for android Video component
```
 <Video
          source={{ uri: 'background' }}
          ref={(ref) => { this.player = ref }}
          rate={1.0}
          volume={1.0}
          muted={true}
          paused={false}
          resizeMode="cover"
          repeat={true}
          playInBackground={false}
          playWhenInactive={false}
          progressUpdateInterval={250.0}
          onLoadStart={this.loadStart}
          onLoad={this.setDuration}
          onProgress={this.setTime}
          onEnd={this.onEnd}
          onError={this.videoError}
          onBuffer={this.onBuffer}
          onTimedMetadata={this.onTimedMetadata}
          style={styles.video}
        />
```

On iOS I use:
```
<Video
          source={require('../Assets/MyScreen/Video/background.mp4')}
          ref={(ref) => { this.player = ref }}
          rate={1.0}
          volume={1.0}
          muted={true}
          paused={false}
          resizeMode="cover"
          repeat={true}
          playInBackground={false}
          playWhenInactive={false}
          progressUpdateInterval={250.0}
          onLoadStart={this.loadStart}
          onLoad={this.setDuration}
          onProgress={this.setTime}
          onEnd={this.onEnd}
          onError={this.videoError}
          onBuffer={this.onBuffer}
          onTimedMetadata={this.onTimedMetadata}
          style={styles.video}
        />
```

On iOS it works as expected.

Any help would be very much appreciated!

**Comment by jonoirwinrsa** [comment]
@Amurmurmur did you run `react-native run-android` after putting the video there?

**Comment by Amurmurmur** [comment]
@jonoirwinrsa Apparently running in the emulator works but once I build and APK release or staging build, the video doesnt get shown at all, its just blank :/

Update:
I build a js bundle for android this way:
```
react-native bundle --platform android --dev false --entry-file index.android.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res/
```

and than just
```
cd android && ./gradlew assembleStaging or assembleRelease
```

**Comment by jonoirwinrsa** [comment]
@Amurmurmur we ran into the same issue but I can't remember exactly how we solved it. I've asked a member of my team who worked on it to comment here if they can... @JameelMukadam

**Comment by Amurmurmur** [comment]
@JameelMukadam @jonoirwinrsa Thank you so much for collaboration 👍  Hoping that you can help to solve this issue.
It think it would be a good idea to document it as well, in an Q&A or FAQ sections or something, once it is resolved (Hopefully!!) :)

**Comment by JameelMukadam** [comment]
@Amurmurmur the way @jonoirwinrsa explained above is correct. have you tried using the res folder or assets folder? i.e. android/app/src/main/res or android/app/src/main/assets

I can't remember which folder it was but was definitely one of those above. it only works on android 5+ so I had to use a different player to play my videos for android <5

**Comment by Amurmurmur** [comment]
@JameelMukadam I have put my background.mp4 in both folders like so:
![screen shot 2017-08-03 at 00 52 24](https://user-images.githubusercontent.com/8608333/28896667-0d6598a8-77e6-11e7-9b7e-879b0ce8af4c.png)
And same code used as I've mentioned above.
I have tried to have source as `{{ uri: 'background' }}` and `{{ uri: 'background.mp4' }}`
If I use `require('../Assets/MyScreen/Video/background.mp4')` the mp4 gets placed under res/drawable folder with the name assets_myscreen_video_background.mp4 given to it.

And obviously when building, an error gets thrown, such as `assets_myscreen_video_background.mp4 is not an xml or png`

**Comment by JameelMukadam** [comment]
@Amurmurmur have you tried putting it directly in the res folder and not the raw folder? and what version of Android are you using?

**Comment by jonoirwinrsa** [comment]
@Amurmurmur sorry for not being much help. 

We did get it working like the above at one point but later used [react-native-vlc-player](https://github.com/ghondar/react-native-vlc-player) instead

**Comment by PetarZo** [comment]
Does anyone know what the folder would be for iOS? 
Android works fine, thank you!

**Comment by rudra0713** [comment]
Hi, I am having a similar issue with android. I have 3 custom images. Each of them renders fine in Android emulator and IOS simulator. However, on an Android device, these images were not showing at first. So I created an assets folder in android/app/src/main/ directory and put my 'img' folder there and ran
`react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res/`

Now two of the images are showing fine but not the third.  The only solution I found on internet was to run that command which I already did and now I am clueless.
